includes/apple-exporter/components/class-body.php

```diff
@@ -440,6 +439,42 @@ public function set_default_style() {
 		);
 	}
 
+	/**
+	 * Determine whether to apply a dropcap style for the component.
+	 * 
+	 * @param \Apple_Exporter\Theme $theme Object that stores theme level dropcap configuration.
+	 * @param string                $html The HTML to check for dropcap conditions. Should be the first paragraph of the post content.
+	 *
+	 * @return boolean
+	 */
+	private function dropcap_determination( $theme, $html ) {
+		// Toggle dropcap determination flag so that this logic applies only to the post's first paragraph.
+		$theme->dropcap_determined = true;
+
+		// Check that the theme is configured to apply dropcap styling.
+		if ( 'yes' !== $theme->get_value( 'initial_dropcap' ) ) {
+			return false;
+		}
+
+		// Check the first character for punctuation.
+		$content    = wp_strip_all_tags( $html );
+		$first_char = mb_substr( $content, 0, 1 );
+		// Regex-planation: \p{P} searchs for punctuation, /u modifier makes it unicode inclusive.
+		if ( preg_match( '/\p{P}$/u', $first_char ) ) {
+			return false;
+		}
+
+		// Check that the content meets the minimum character number.
+		$num_chars = mb_strlen( $content );
+		if ( 'yes' !== $theme->get_value( 'dropcap_minimum_opt_out' )
+			&& $num_chars < (int) $theme->get_value( 'dropcap_minimum' )
```

Comment:
This should be inverted - it should only apply the dropcap if it is _at least_ the number of characters specified in the minimum. This is to avoid a situation where the dropcap looks janky because there isn't enough text to wrap around it.

Reply:
This is one of the conditions that would prevent the dropcap from being applied.  So if the number of characters is less than the dropcap minimum then the whole function (`dropcap_determination()`) returns `false` and `$this->set_default_style()` is called above on line 303 (applying normal text styles).

If the number of characters matches or exceeds the minimum it passes over the condition and the whole function (`dropcap_determination()`) returns `true` and `$this->set_initial_dropcap_style()` is called up on line 301 (applying dropcap text styles).

Reply:
Right, this is inverted logic - if the condition is true then it returns false, and doesn't apply the dropcap.

Reply:
@kevinfodness Oh, I think I understand.  You're suggesting I flip the conditional and then return true (or the in this case the `apply_filter()` bit) inside the conditional itself?

Reply:
Ah nevermind, that's not going to work with the `$use_dropcap` variable being part of the filter.  Lemme take another look and see if there's another way to simplify this.

Reply:
Basically it should look something like this:
```php
$use_dropcap = true;
if ( $reason_why_we_shouldnt_use_dropcap_1 ) {
    $use_dropcap = false;
} elseif ( $reason_why_we_shouldnt_use_dropcap_2 ) {
    $use_dropcap = false;
}
return apply_filters( // etc
```

Reply:
Oh right, `elseif` will break out of the whole string of conditions.  Was trying to wrap everything in a while loop which was resulting in two filter returns.  That's cleaner/clearer.

Still not sure how to invert that last conditional checking for min characters if we're looking for it to return false but I'll have another go at refactoring it.